hydra_agent/redis_core/graphutils_operations.py

```diff
@@ -192,5 +211,36 @@ def get_resource(self, url: str) -> dict:
 
         return resource
 
+    def embedded_resource(self, parent_id: str, parent_type: str,
+                          discovered_url: str) -> str:
+        """Checks for existance of discovered resource and creates links
+        for embedded resources inside other resources properties
+        :parent_id: Resource ID for the parent node that had this reference
+        :parent_type: Resource Type for the parent node that had this reference
+        :discovered_url: URL Reference for resource found inside a property
+        """
+        resource = self.get_resource(discovered_url)
+        if resource is None:
+            response = self.session.get(discovered_url)
```

Comment:
As written above, this line should call a method from another class, not make this class to instantiate its own Session.